### PR TITLE
issue 964 properly impl getEntityTypeByLabel

### PR DESCRIPTION
**Pull request by AndrewZWood**
Fixes https://github.com/DSpace/dspace-angular/issues/964

### CODE REVIEW

File: src/app/core/data/entity-type.service.ts [hunk 1]
```diff
@@ -88,30 +88,10 @@ export class EntityTypeService extends DataService<ItemType> {
...
-      'OrgUnit',
-      'Journal',
-      'JournalVolume',
-      'JournalIssue',
-      'DataPackage',
-      'DataFile',
-    ].indexOf(label);
-
-    return this.findById((index + 1) + '');
+    return this.findByHref(endPoint + '/label/' + label);
```
Comment (artlowel) [inline src/app/core/data/entity-type.service.ts:95]: `endPoint` will only get a value if `this.halService.getEndpoint()` is synchronous. That won't be the case if it isn't already cached, or if the cache time runs out.

 What you want is something like:
```typescript
return this.halService.getEndpoint(this.linkPath).pipe(
  take(1),
  switchMap((endPoint: string) =>
    this.findByHref(endPoint + '/label/' + label))
);
```

That way `this.findByHref` will wait for the result of `this.halService.getEndpoint` even if it isn't synchronous

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @AndrewZWood!

Theres an issue with the implementation. See my inline comment.

File: src/app/core/data/entity-type.service.ts (hunk 1 above)
Comment (AndrewZWood) [reply src/app/core/data/entity-type.service.ts:95]: That neat.  Didn't know I could return like that. Will Update and test this afternoon 9 ( EST ).

Comment (AndrewZWood) [reply src/app/core/data/entity-type.service.ts]: Updated code and tested with local data: https://github.com/DSpace/dspace-angular/pull/965/commits/06bb2130783c2c042b1ea7698809fd3f0bec8124

**Review by artlowel — APPROVED** [review]
Thanks @AndrewZWood I can verify that your code works now. 

However it has brought a bug on the rest side to light. The entity types retrieved in this way can't be parsed, because they're lacking a _links section. I created an issue for it.

So this PR should not be merged until that issue is fixed on the backend

**Comment by AndrewZWood** [comment]
@artlowel Already have an issue and PR for what you're describing: https://github.com/DSpace/DSpace/issues/3078

**Review by tdonohue — APPROVED** [review]
👍  I tested this alongside https://github.com/DSpace/DSpace/pull/3079 (corresponding REST PR) and they work great!  Thanks @AndrewZWood ! 

I'll merge both together & ensure the REST PR is deployed to the demo site.

**Pull request merged**